### ISSUE TITLE
Path issues with a mac (capitalisation)

### ISSUE THREAD

**Issue by NobbZ** [exercism/cli#637]
In exercism/exercism.io#3820 we discovered that when beeing in `~/exercism/javascript/hello-world` canceled submission because of not beeing in the workspace, as the configured workspace was `~/Exercism` (capital `E`!). Downloading exercises worked and files were accessible at the aforementioned uncapitalized path.

It seems as if we need to normalize pathes on mac as well similar how it is planned or already done for windows.

**Comment by nywilken** [comment]
@nobbz thanks for calling this out. We have not yet fixed this issue, but it’s on our radar. 

File paths on configure are titlized but we don’t normalize for paths on submit.

**Comment by kytrinyx** [comment]
@NobbZ, yes! I completely agree. We need to make sure that we pay attention to what is configured and ensure that _that_ is what we use to submit.

**Comment by kytrinyx** [comment]
@NobbZ If you have a moment, would you help fleshing out the reproduction steps we need to ensure that we fix this correctly both for the case sensitive and case insensitive platforms?

The reported case was:

- workspace directory is capitalized
- CLI is configured with capitalized directory name (matches actual name)
- exercise is submitted using lowercase for workspace

It sounds like we want two different outcomes, depending on whether or not the filesystem is case sensitive or not.

1. Case insensitive file system: the exercise is submitted
1. Case sensitive file system: we get an error saying that the file that is being submitted it outside the workspace.

My mac is case sensitive, but only to a certain degree:

If I have a directory named `test`, then:

- `mkdir Test` will fail with `File exists`
- `ls test` will show `test`
- `ls te<tab>` will auto-complete
- `ls Test` will show `Test`
- `ls Te<tab>` will not auto-complete

So fundamentally, Mac is case insensitive in that it will not allow you to have two directories where the name differs only in capitalization.

I think that the right way to move forward in this is to have two separate test files, one that compiles on linux, and the other that compiles for non-linux.

**Comment by kytrinyx** [comment]
Note from @NobbZ:

> I googled a bit in the meantime, perhabs the `HasFilepathPrefix` function from this internal `dep` module can help you? https://github.com/golang/dep/blob/master/internal/fs/fs.go

This looks exactly like what we need.

**Comment by kotp** [comment]
Bumping this issue, as quite a few "support requests" come about regarding this issue.
